app/src/main/java/org/mozilla/focus/utils/LiveDataEvent.kt

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.utils
+
+/**
+ * An object assigned a LiveData instance to send a stateless event. See
+ * [org.mozilla.focus.utils.UserClearDataEvent] for example usage.
+ *
+ * This implementation is inspired by:
+ *   https://medium.com/google-developers/livedata-with-snackbar-navigation-and-other-events-the-singleliveevent-case-ac2622673150
+ */
+
+class LiveDataEvent {
+    var hasBeenHandled = false
```

Comment:
Unless I'm not understanding your comment correctly, it's not necessarily restricting to one observer. The purpose is to reset the event when the event occurs. The name may be too generic but the intention was singular implication, single consumption event. Otherwise, no point having this event - just use boolean value with livedata

Reply:
It's entirely possible that I'm misunderstanding the code, so please correct me if I'm wrong.  If I understand correctly, this `hasBeenHandled` concept prevents situations where a fragment is destroyed and recreated, receives a cached event from the LiveData, then clears again based on this outdated event.  That makes sense as a solution, but it means that if there were ever more than one observer subscribed to these events only one of them would act (and I'm not at all certain which one).  Having a globally accessible bus that has an implicit rule that it can only be observed by one object seems like a landmine to me.

I'm not super familiar with LiveData, but if I were approaching this with Rx I would use a different type of observable that didn't cache the last seen value (this would also require some lifecycle tweaks).

If you don't think this is a big deal, I'm fine with documenting the behavior and moving on.  It just seems like we're fighting LiveData's behavior, and I wonder if it's not the best tool for this job.

Reply:
You're correct sir, AFAIK LiveDatas are bound to Fragment's/Activity's lifecycle. Receiving a cached (latest) event when fragment is in onDestroyView (vs. onDestroy) / onCreateView (vs. onCreate) (AFAIK this happens quite often as opposed to Activity lifecycle) should be an expected behaviour, no? I can see your concern though:

```An observer added with a Lifecycle will be automatically removed if the corresponding Lifecycle moves to DESTROYED state. This is especially useful for activities and fragments where they can safely observe LiveData and not worry about leaks: they will be instantly unsubscribed when they are destroyed. ```

I think the context here is to clear BrowserFragment's webview IFF the instance exists. So wouldn't you say it's a explicit relationship between the two webview instances (Settings and BrowserFragment)? There is something wrong with where I am initializing the live data observer (what Michael pointed out above) which is the essence of using livedata vs. Rx